baseapp/baseapp.go

```diff
@@ -705,12 +705,13 @@ func (app *BaseApp) cacheTxContext(ctx sdk.Context, txBytes []byte) (sdk.Context
 	return ctx.WithMultiStore(msCache), msCache
 }
 
-func (app *BaseApp) preBlock(req *abci.RequestFinalizeBlock) error {
+func (app *BaseApp) preBlock(req *abci.RequestFinalizeBlock) ([]abci.Event, error) {
+	var events []abci.Event
 	if app.preBlocker != nil {
 		ctx := app.finalizeBlockState.Context()
```

Comment:
```suggestion
		ctx := app.finalizeBlockState.Context().WithEventManager(sdk.NewEventManager())
```

Then below, instead of `events = rsp.Events`, you can use `events = ctx.EventManager().ABCIEvents()`.

This makes it so that you don't have to change `ResponsePreBlock` or the `func (m *Manager) PreBlock(...)`. It also better matches what the SDK has in `main`.